src/pyats/contrib/plugins/topoup_plugin/topoup.py

```diff
@@ -0,0 +1,139 @@
+                         help='Enable/Disable checking for topology up pre job execution')
+
+        # -connection_check_timeout
+        # --connection_check_timeout
+        grp.add_argument(*connection_check_timeout,
+                         dest='connection_check_timeout',
+                         action='store',
+                         default=120,
+                         help='Total time allowed for checking devices connectivity')
+
+        # -connection_check_interval
+        # --connection_check_interval
+        grp.add_argument(*connection_check_interval,
+                         dest='connection_check_interval',
+                         action='store',
+                         default=10,
+                         help='Time to sleep between device connectivity checks')
+
+        return grp
+
+
+    def pre_job(self, task):
+        '''Loop over all the topology devices and make sure they are up and running
+           before executing the test script.
+        '''
+
+        # Set the timers
+        start_time = time()
+        timeout = self.runtime.args.connection_check_timeout
+        interval = self.runtime.args.connection_check_interval
+
+        log.info("Connectivity check timeout is '{timeout}' and "
+            "connectivity check interval is '{interval}'".format(timeout=timeout, interval=interval))
+
+        # Looping over devices and make sure they are up
+        pcall_output = pcall(device_connect,
+            ckwargs = {'start_time': start_time, 'timeout': timeout, 'interval': interval},
+            ikwargs = [{'device':self.runtime.testbed.devices[dev]} for dev in self.runtime.testbed.devices])
+
+        if not (pcall_output[0] and pcall_output[1]):
+            # Terminate testscript
+            raise Exception ("Not all the testbed devices are up and ready")
+
+        return
+
+
+def device_connect(device, start_time, timeout, interval):
+    '''Try to connect to the device and if fails, sleep for interval seconds and retry
+       till the timeout is reached
+
+        Args:
+            device ('obj'): device to use
+            start_time ('int'): Current time to calculate the timeout, seconds
+            timeout ('int'): Timeout value when reached exit even if failed, seconds
+            interval ('int'): Sleep time between retries, seconds
+
+        Returns:
+            result(`bool`): Device is successfully connected
+
+        Raises:
+            None
+
+    '''
+
+    time_difference = time() - start_time
+
+    while (time() - start_time) < float(timeout):
+
+        try:
+            # Connect to the device
+            device.connect()
+            log.info("Successfully connected to '{device}'".format(device=device))
+
+            # Return the pcall call with True
+            return True
+
+        except:
+            # Not ready sleep and retry
+            log.info("Sleeping for '{interavl}' seconds and retry, remaining time {remaining_time}".format(
```

Comment:
The message should say it failed and typo at interval

Reply:
Done